opm/simulators/wells/BlackoilWellModelConstraints.cpp

```diff
@@ -483,7 +483,7 @@ actionOnBrokenConstraints(const Group& group,
     std::string ss;
     switch (group_limit_action.allRates) {
     case Group::ExceedAction::NONE: {
-        if (oldControl != newControl && oldControl != Group::ProductionCMode::NONE) {
+        if (oldControl != newControl) {
```

Comment:
I think the change is positive. I do not understand why we need to check `oldControl` to decide whether we can switch to a group control.

Reply:
With more investigation, I think the fix is correct, while I am not convinced we are managing the actions exceeding group constraint properly as a whole. 

But removing `oldControl != Group::ProductionCMode::NONE` is sensible any way.

Reply:
Great, thanks for reviewing this!

Reply:
For example,  we probably should not include all the actions inside this condition.  In the parsing stage, we should generate the proper actions for `oil, water, gas and liquid` properly, because the item 7 and item 11-13 and item 2 should have combined effects. We should handle it in the parsing stage.  The following code might have problems if we defined item 7 (not NONE), the definition in item 11-13 might be ignored (which should override item 7 actually).  For sure, the current way of working will have big problem to be extended to other actions. 
(It is complicated and I might read things wrong. But leaving some comments in case we will revisit this part in the future development)

```c++
    switch (group_limit_action.allRates) {
    case Group::ExceedAction::NONE: {
```

Reply:
I think you are right in that this needs to be changed in to support other actions than 'RATE', but for now it appears that all combinations of NONE or RATE in items 7,11,12,13 work as expected (?)

Reply:
Too occupied to go check the details, while not sure whether it will work if we specifies item 7 with RATE, while using `NONE` in some of the items 11-13 will work. 

For example, (item 2 is 'NONE`) if we use `RATE` for item 7, and use `NONE` for all of the items 11-13, then it should be RATE action for oil constraint (not considering the constraints after item 13) and NONE for water, gas and liquid.  

Furthermore, when we using `allRates` in the parser side, it is difficult to do something specifically for `oil`. 

I can be wrong though, testing might show otherwise.

Reply:
It is not clear to me whether we can use `NONE` in item 11-13 to override non-NONE actions in item 7 either.

Reply:
Did a quick test to confirm, and at least this seems to work as expected ([test.pdf](https://github.com/user-attachments/files/15956287/test.pdf)):
```
GCONPROD
  'PLAT-A' 'NONE' 3900.0 100.0 100.0 100.0 RATE 3* NONE NONE NONE /
/
```

Reply:
Thanks for checking. It will be interesting to see the manner how the actions get enforced.  My hypothesis was wrong.

Reply:
It seems the way it works is by never adding the constraints in the first place if overridden by 'NONE' in items 11-13 (but I could be wrong.. :) )

Reply:
I think you are correct. That is the only way it can work, otherwise the simulator side code will not work for the specification you specified.

Reply:
Last comment here is that we have `allRates` and the `water`, `gas` and `liquid` (they considered the item 7 and item 11-13). We use `allRates` for the `oil` and the others for themselves respectively.  Probably we can using `oil` instead of `allRates` here.  And in the simulator code, we can avoid the current way of using `allRates` since we will not have it there.